lib/db.js

```diff
@@ -479,19 +482,50 @@ module.exports = (
       uaDeviceType: token.uaDeviceType,
       lastAccessTime: token.lastAccessTime,
       createdAt: token.createdAt
-    }]
+    }
     let sessionTokens = []
+    let matchedTokens = []
+    const ipHash = ip && crypto.createHash('sha256').update(ip).digest('base64')
```

Comment:
>  I would expect doing the ip -> location lookup to be quite fast

FWIW, my expectations don't *always* match reality, it would be worth doing a quick benchmark to sanity-check this assertion :-)